data/models.py

```diff
@@ -17,17 +18,155 @@ class DDSUserCredential(models.Model):
     """
     DDS Credentials for bespin users
     """
+    endpoint = models.ForeignKey(DDSEndpoint, on_delete=models.CASCADE, null=False)
     user = models.OneToOneField(settings.AUTH_USER_MODEL, null=False)
     token = models.CharField(max_length=32, blank=False, unique=True)
 
 
-class DDSResource(models.Model):
-    owner = models.ForeignKey(settings.AUTH_USER_MODEL, null=False)
-    project_id = models.UUIDField(blank=False)
-    path = models.TextField(null=False)
+class Workflow(models.Model):
+    """
+    Name of a workflow that will apply some processing to some data.
+    """
+    name = models.CharField(max_length=255, blank=False)
+
+    def __unicode__(self):
+        return self.name
 
-    class Meta:
-        unique_together = (
-            ('owner', 'project_id', 'path'),
-        )
 
+class WorkflowVersion(models.Model):
+    """
+    Specific version of a Workflow.
+    """
+    workflow = models.ForeignKey(Workflow, on_delete=models.CASCADE, related_name='versions')
+    object_name = models.CharField(max_length=255, null=True)
+    created = models.DateTimeField(auto_now_add=True, blank=False)
+    version = models.CharField(max_length=32, blank=False)
+    url = models.URLField()
```

Comment:
Is there a way we could detect or enforce a specific version? I think it should be hard to mutate a workflow version. As it stands, the URL could be changed, and the target of the URL could be modified without detection.

Perhaps the URL field should be marked [non-editable](https://docs.djangoproject.com/en/1.10/ref/models/fields/#editable). While I'm not sure that will prevent direct model modification, it should prevent forms and serializers from changing it.

Additionally, it might be a good idea to include a checksum for the target of the url.